app/services/search_query_parser.rb

```diff
@@ -0,0 +1,22 @@
+class SearchQueryParser
+  API_URL = ENV['TARIFF_QUERY_SEARCH_PARSER_URL']
+
+  def initialize
+    @client = Faraday.new(
+      url: API_URL,
+      headers: {'Content-Type' => 'application/json'}
+      )
+  end
+
+  def get_tokens(search_query_term)
+    api_response = @client.get("tokens/#{search_query_term}")
+
+    response_data = api_response.success? ? JSON.parse(api_response.body) : {}
```

Comment:
I think we should be handling the errors within this Parser rather then relying on the consumer of `SearchQueryParser` to in turn then parse the output hash from the Parser.

I'd suggest instead including the Faraday Errors handler - any unexpected response codes and it will trigger an exception - see

https://github.com/trade-tariff/trade-tariff-frontend/blob/main/app/services/client_builder.rb#L29

The method return value could then just be `response_data` - if `success` was false then an exception would be raised so is redundant, as is `status`

Reply:
I've also realised the JSON parsing can be handled automatically if you configure the Farady JSON response middleware when setting up the connection?

Reply:
yep, I've changed the way Faraday is initialized, and we don't need it anymore.